pl/lessons/libraries/benchee.md

```diff
@@ -0,0 +1,335 @@
+
+Istnieją również inne dostępne statystyki, ale te cztery są często najbardziej użyteczne i powszechnie używane do analizy porównawczej, dlatego są wyświetlane w domyślnym formaterze. Więcej informacji na temat innych dostępnych metryk można znaleźć w dokumentacji [hexdocs](https://hexdocs.pm/benchee/Benchee.Statistics.html#statistics/1).
+
+# Konfiguracja
+
+Jedną z najlepszych części Benchee są wszystkie dostępne opcje konfiguracji. Zaczniemy do podstaw, ponieważ nie wymagają przykładów kodu, a następnie pokażemy, jak wykorzystać jedną z najlepszych funkcji Benchee - wejść.
+
+## Podstawy
+
+Benchee ma wiele opcji konfiguracyjnych. W najbardziej popularnym interfejsie `Benchee.run/2`, są one przekazywane jako drugi argument w formie listy słów kluczowych:
+
+```elixir
+Benchee.run(%{"example function" => fn -> "hi!" end}, [
+  warmup: 4,
+  time: 10,
+  inputs: nil,
+  parallel: 1,
+  formatters: [&Benchee.Formatters.Console.output/1],
+  print: [ 
+    benchmarking: true,
+    configuration: true,
+    fast_warning: true
+  ],
+  console: [
+    comparison: true,
+    unit_scaling: :best
+  ]
+])
+```
+
+Dostępne są następujące opcje (także udokumentowane w [hexdocs](https://hexdocs.pm/benchee/Benchee.Configuration.html#init/1)).
+
+* **warmup** - czas w sekundach, dla którego powinien być uruchomiony scenariusz porównawczy bez czasów pomiarowych, zanim zaczną się rzeczywiste pomiary. To symuluje "ciepłe" działanie systemu. Domyślne ustawienie to 2 sekundy.
+* **time** - czas w sekundach, jak długo powinien być uruchamiany i mierzony każdy indywidualny scenariusz porównawczy. Domyślnie 5 sekund.
+* **inputs** - mapa z łańcuchami reprezentującymi nazwę wejściową jako klucze i rzeczywiste dane wejściowe jako wartości. Domyślnie `nil`. W dalszej części omówimy to szczegółowo.
+* **parallel** - liczba procesów używanych do testu wydajności danej funkcjonalności. Więc jeśli ustawisz 'parallel: 4', wtedy zostaną stworzone cztery procesy, które wykonują tę samą funkcję dla danego `time`. Kiedy skończą się, na następnej funkcji zostaną uruchomione cztery nowe procesy. Daje to więcej danych w tym samym czasie, ale także powoduje obciążenie systemu zakłócające wyniki testów. Może to być przydatne do symulacji systemu pod obciążeniem, które jest czasem pomocne, ale powinno być stosowane z pewną ostrożnością, ponieważ może to wpływać na wyniki w nieprzewidywalny sposoby. Domyślnie 1 (co oznacza brak równoległej realizacji).
+* **formatters** - lista funkcji formatyzacji, które chcesz uruchomić, aby uzyskać wyniki testów porównawczych pakietu przy użyciu polecenia `Benchee.run / 2`. Funkcje muszą zaakceptować jeden argument (czyli pakiet porównawczy z wszystkimi danymi), a następnie użyć go do produkcji danych wyjściowych. Domyślny formater to: `Benchee.Formatters.Console.output / 1`. Omówimy to w dalszej części.
```

Comment:
There are some extra space between the functions' names and their arities. Was this made purposely?
This question also applies to [line 170](https://github.com/elixirschool/elixirschool/pull/1168/files#diff-e918698f91a2d5fb34dbadeedc5529bdR170).